core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -326,6 +329,9 @@ public int read(ByteBuffer byteBuffer, int off, int len) throws IOException {
     byteBuffer.position(off).limit(off + toRead);
     mCurrentChunk.readBytes(byteBuffer);
     mPos += toRead;
+    if (mPos == mLength) {
+      closeDataReader();
```

Comment:
add javadoc for tricky improvement, so in the future we will pay attention when touching this piece of code.

```suggestion
      // reference the performance issue on github
      closeDataReader();
```